apps/nextjs-pages-router/src/pages/[[...path]].tsx

```diff
@@ -30,22 +30,35 @@ export async function getStaticPaths(): Promise<
   }
 }
 
-type Props = MakeswiftPageProps
+export type PageProps = MakeswiftPageProps & {
+  previewMode: boolean
+  locale: string | undefined
+}
 
-export async function getStaticProps(
-  ctx: GetStaticPropsContext<ParsedUrlQuery>,
-): Promise<GetStaticPropsResult<Props>> {
-  const path = '/' + (ctx.params?.path ?? []).join('/')
+export async function getStaticProps({
+  params,
+  previewData,
+  locale,
+}: GetStaticPropsContext<ParsedUrlQuery>): Promise<
+  GetStaticPropsResult<PageProps>
+> {
+  const path = '/' + (params?.path ?? []).join('/')
   const snapshot = await client.getPageSnapshot(path, {
-    siteVersion: Makeswift.getSiteVersion(ctx.previewData),
-    locale: ctx.locale,
+    siteVersion: Makeswift.getSiteVersion(previewData),
+    locale,
   })
 
   if (snapshot == null) return { notFound: true }
 
-  return { props: { snapshot } }
+  return {
+    props: {
+      snapshot,
+      previewMode: Makeswift.getPreviewMode(previewData),
```

Comment:
Using a new method for deriving `previewMode` from `previewData`, see below.

Reply:
@migueloller The result of this method is what's passed as props to the `App` [here](https://github.com/makeswift/makeswift/pull/836/files#diff-7726b5511392f97ec99ec7c5c5727dda11bd3d5c8d01621b2e11dabb4c9cfa20R7), isn't it?

Reply:
Yes, I see this now. We can move this discussion to a separate thread re: `_app` because, while I won't go as far as calling it an anti-pattern, I consider this sort of stuff (reading props from a page on `_app`) undesirable in Next.js Pages Router.